JDK 17 ARM64 doesnt work - Java cannot run in  AWS Fargate

### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  ```text
uname -a
Linux ip-.x.x.x.x.compute.internal 4.14.291-218.527.amzn2.aarch64 #1 SMP Fri Aug 26 09:54:28 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
  ```

</details>


### What Operating System are you using (both controller, and any agents involved in the problem)?

  ```text
uname -a
Linux ip-.x.x.x.x.compute.internal 4.14.291-218.527.amzn2.aarch64 #1 SMP Fri Aug 26 09:54:28 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
  ```

### Reproduction steps

# Try 1
1. On an amazon linux 2 machine with docker installed on ARM64 in AWS (IE t4g.medium)
2. `docker pull jenkins/agent:3071.v7e9b_0dc08466-1-jdk17`
3. `docker run -it --entrypoint /bin/bash  jenkins/agent:3071.v7e9b_0dc08466-1-jdk17`
   a. Run `java` within the container, error happens below:
   `bash: /opt/java/openjdk/bin/java: cannot execute binary file: Exec format error`


# Try 2
Also tried the ARM64 specific image:
`docker run -it --entrypoint /bin/bash  jenkins/agent:3071.v7e9b_0dc08466-1-jdk17@sha256:bc386f73b3654d7ccaa73060ee66453be35795eb48d9d670d0eaeaa38e16688e`

`bash: /opt/java/openjdk/bin/java: cannot execute binary file: Exec format error`

# Try3
If I use the preview image it works...
`docker run -it --entrypoint /bin/bash  jenkins/agent:latest-jdk17-preview`

<details>
  <summary>Command Output</summary>

```
jenkins@a44fdd3f66c0:~$ java
Usage: java [options] <mainclass> [args...]
           (to execute a class)
   or  java [options] -jar <jarfile> [args...]
           (to execute a jar file)
   or  java [options] -m <module>[/<mainclass>] [args...]
       java [options] --module <module>[/<mainclass>] [args...]
           (to execute the main class in a module)
   or  java [options] <sourcefile> [args]
           (to execute a single source-file program)

 Arguments following the main class, source file, -jar <jarfile>,
 -m or --module <module>/<mainclass> are passed as the arguments to
 main class.

 where options include:

    -cp <class search path of directories and zip/jar files>
    -classpath <class search path of directories and zip/jar files>
    --class-path <class search path of directories and zip/jar files>
                  A : separated list of directories, JAR archives,
                  and ZIP archives to search for class files.
    -p <module path>
    --module-path <module path>...
                  A : separated list of directories, each directory
                  is a directory of modules.
    --upgrade-module-path <module path>...
                  A : separated list of directories, each directory
                  is a directory of modules that replace upgradeable
                  modules in the runtime image
    --add-modules <module name>[,<module name>...]
                  root modules to resolve in addition to the initial module.
                  <module name> can also be ALL-DEFAULT, ALL-SYSTEM,
                  ALL-MODULE-PATH.
    --enable-native-access <module name>[,<module name>...]
                  modules that are permitted to perform restricted native operations.
                  <module name> can also be ALL-UNNAMED.
    --list-modules
                  list observable modules and exit
    -d <module name>
    --describe-module <module name>
                  describe a module and exit
    --dry-run     create VM and load main class but do not execute main method.
                  The --dry-run option may be useful for validating the
                  command-line options such as the module system configuration.
    --validate-modules
                  validate all modules and exit
                  The --validate-modules option may be useful for finding
                  conflicts and other errors with modules on the module path.
    -D<name>=<value>
                  set a system property
    -verbose:[class|module|gc|jni]
                  enable verbose output for the given subsystem
    -version      print product version to the error stream and exit
    --version     print product version to the output stream and exit
    -showversion  print product version to the error stream and continue
    --show-version
                  print product version to the output stream and continue
    --show-module-resolution
                  show module resolution output during startup
    -? -h -help
                  print this help message to the error stream
    --help        print this help message to the output stream
    -X            print help on extra options to the error stream
    --help-extra  print help on extra options to the output stream
    -ea[:<packagename>...|:<classname>]
    -enableassertions[:<packagename>...|:<classname>]
                  enable assertions with specified granularity
    -da[:<packagename>...|:<classname>]
    -disableassertions[:<packagename>...|:<classname>]
                  disable assertions with specified granularity
    -esa | -enablesystemassertions
                  enable system assertions
    -dsa | -disablesystemassertions
                  disable system assertions
    -agentlib:<libname>[=<options>]
                  load native agent library <libname>, e.g. -agentlib:jdwp
                  see also -agentlib:jdwp=help
    -agentpath:<pathname>[=<options>]
                  load native agent library by full pathname
    -javaagent:<jarpath>[=<options>]
                  load Java programming language agent, see java.lang.instrument
    -splash:<imagepath>
                  show splash screen with specified image
                  HiDPI scaled images are automatically supported and used
                  if available. The unscaled image filename, e.g. image.ext,
                  should always be passed as the argument to the -splash option.
                  The most appropriate scaled image provided will be picked up
                  automatically.
                  See the SplashScreen API documentation for more information
    @argument files
                  one or more argument files containing options
    -disable-@files
                  prevent further argument file expansion
    --enable-preview
                  allow classes to depend on preview features of this release
To specify an argument for a long option, you can use --<name>=<value> or
--<name> <value>.
```
</details

### Expected Results

When running java it should print the java help and load the runtime correctly:
```
jenkins@169274143b43:~$ java
Usage: java [options] <mainclass> [args...]
           (to execute a class)
   or  java [options] -jar <jarfile> [args...]
           (to execute a jar file)
   or  java [options] -m <module>[/<mainclass>] [args...]
       java [options] --module <module>[/<mainclass>] [args...]
           (to execute the main class in a module)
   or  java [options] <sourcefile> [args]
           (to execute a single source-file program)

 Arguments following the main class, source file, -jar <jarfile>,
 -m or --module <module>/<mainclass> are passed as the arguments to
 main class.

 where options include:

    -cp <class search path of directories and zip/jar files>
    -classpath <class search path of directories and zip/jar files>
    --class-path <class search path of directories and zip/jar files>
                  A : separated list of directories, JAR archives,
                  and ZIP archives to search for class files.
    -p <module path>
    --module-path <module path>...
                  A : separated list of directories, each directory
                  is a directory of modules.
    --upgrade-module-path <module path>...
                  A : separated list of directories, each directory
                  is a directory of modules that replace upgradeable
                  modules in the runtime image
    --add-modules <module name>[,<module name>...]
                  root modules to resolve in addition to the initial module.
                  <module name> can also be ALL-DEFAULT, ALL-SYSTEM,
                  ALL-MODULE-PATH.
    --enable-native-access <module name>[,<module name>...]
                  modules that are permitted to perform restricted native operations.
                  <module name> can also be ALL-UNNAMED.
    --list-modules
                  list observable modules and exit
    -d <module name>
    --describe-module <module name>
                  describe a module and exit
    --dry-run     create VM and load main class but do not execute main method.
                  The --dry-run option may be useful for validating the
                  command-line options such as the module system configuration.
    --validate-modules
                  validate all modules and exit
                  The --validate-modules option may be useful for finding
                  conflicts and other errors with modules on the module path.
    -D<name>=<value>
                  set a system property
    -verbose:[class|module|gc|jni]
                  enable verbose output for the given subsystem
    -version      print product version to the error stream and exit
    --version     print product version to the output stream and exit
    -showversion  print product version to the error stream and continue
    --show-version
                  print product version to the output stream and continue
    --show-module-resolution
                  show module resolution output during startup
    -? -h -help
                  print this help message to the error stream
    --help        print this help message to the output stream
    -X            print help on extra options to the error stream
    --help-extra  print help on extra options to the output stream
    -ea[:<packagename>...|:<classname>]
    -enableassertions[:<packagename>...|:<classname>]
                  enable assertions with specified granularity
    -da[:<packagename>...|:<classname>]
    -disableassertions[:<packagename>...|:<classname>]
                  disable assertions with specified granularity
    -esa | -enablesystemassertions
                  enable system assertions
    -dsa | -disablesystemassertions
                  disable system assertions
    -agentlib:<libname>[=<options>]
                  load native agent library <libname>, e.g. -agentlib:jdwp
                  see also -agentlib:jdwp=help
    -agentpath:<pathname>[=<options>]
                  load native agent library by full pathname
    -javaagent:<jarpath>[=<options>]
                  load Java programming language agent, see java.lang.instrument
    -splash:<imagepath>
                  show splash screen with specified image
                  HiDPI scaled images are automatically supported and used
                  if available. The unscaled image filename, e.g. image.ext,
                  should always be passed as the argument to the -splash option.
                  The most appropriate scaled image provided will be picked up
                  automatically.
                  See the SplashScreen API documentation for more information
    @argument files
                  one or more argument files containing options
    -disable-@files
                  prevent further argument file expansion
    --enable-preview
                  allow classes to depend on preview features of this release
To specify an argument for a long option, you can use --<name>=<value> or
--<name> <value>.

```

### Actual Results

```
jenkins@b5f5793e03a8:~$ java
bash: /opt/java/openjdk/bin/java: cannot execute binary file: Exec format error
```

### Anything else?

What is the difference between preview builds and  versioned builds?

Overall I am trying to use the docker-inbound-agent on AWS Fargate ARM, but hitting issues with Java not being the correct runtime in the ARM image.

@gounthar  - sorry for not being clear.  The `--platform` has to match one of the OS/Arch of the [image](https://hub.docker.com/layers/library/eclipse-temurin/17.0.4.1_1-jdk-focal/images/sha256-1eb62dfeec415d857ce5a03230ac5d45dcb7ddfcbb2cee5525220618f79db2e1?context=explore) we are using and so the only option was to set like [this](https://github.com/carpnick/docker-agent/commit/6f2de7aa1762d20f9a22afd8281c2c56706d6316).  That yielded the same error [result](https://ci.jenkins.io/job/Packaging/job/docker-agent/view/change-requests/job/PR-310/4/consoleFull):


<details>
  <summary>Jenkins Error</summary>

  ```text
22:00:51  #43 [debian_jdk17 arm32-jre-build 2/2] RUN jlink   --add-modules ALL-MODULE-PATH   --strip-java-debug-attributes   --no-man-pages   --no-header-files   --compress=2   --output /javaruntime
22:00:51  #43 12.03 Error: java.nio.file.FileSystemException: /opt/java/openjdk/jmods: Value too large for defined data type
22:00:51  #43 12.03 java.nio.file.DirectoryIteratorException: java.nio.file.FileSystemException: /opt/java/openjdk/jmods: Value too large for defined data type
22:00:51  #43 12.04 	at java.base/sun.nio.fs.UnixDirectoryStream$UnixDirectoryIterator.readNextEntry(UnixDirectoryStream.java:169)
22:00:51  #43 12.04 	at java.base/sun.nio.fs.UnixDirectoryStream$UnixDirectoryIterator.hasNext(UnixDirectoryStream.java:198)
22:00:51  #43 12.04 	at java.base/jdk.internal.module.ModulePath.scanDirectory(ModulePath.java:275)
22:00:51  #43 12.04 	at java.base/jdk.internal.module.ModulePath.scan(ModulePath.java:232)
22:00:51  #43 12.04 	at java.base/jdk.internal.module.ModulePath.scanNextEntry(ModulePath.java:190)
22:00:51  #43 12.04 	at java.base/jdk.internal.module.ModulePath.find(ModulePath.java:154)
22:00:51  #43 12.04 	at jdk.jlink/jdk.tools.jlink.internal.JlinkTask.newModuleFinder(JlinkTask.java:468)
22:00:51  #43 12.04 	at jdk.jlink/jdk.tools.jlink.internal.JlinkTask.initJlinkConfig(JlinkTask.java:387)
22:00:51  #43 12.04 	at jdk.jlink/jdk.tools.jlink.internal.JlinkTask.run(JlinkTask.java:271)
22:00:51  #43 12.04 	at jdk.jlink/jdk.tools.jlink.internal.Main.run(Main.java:55)
22:00:51  #43 12.04 	at jdk.jlink/jdk.tools.jlink.internal.Main.main(Main.java:33)
22:00:51  #43 12.04 Caused by: java.nio.file.FileSystemException: /opt/java/openjdk/jmods: Value too large for defined data type
22:00:51  #43 12.04 	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:100)
22:00:51  #43 12.04 	at java.base/sun.nio.fs.UnixException.asIOException(UnixException.java:115)
22:00:51  #43 12.04 	at java.base/sun.nio.fs.UnixDirectoryStream$UnixDirectoryIterator.readNextEntry(UnixDirectoryStream.java:168)
22:00:51  #43 12.04 	... 10 more
22:00:53  #43 ERROR: process "/bin/sh -c jlink   --add-modules ALL-MODULE-PATH   --strip-java-debug-attributes   --no-man-pages   --no-header-files   --compress=2   --output /javaruntime" did not complete successfully: exit code: 4
22:00:53  
22:00:53  #42 [archlinux_jdk11 stage-1 4/8] RUN pacman -Syu git git-lfs --noconfirm
  ```

</details>

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -34,12 +34,12 @@ FROM eclipse-temurin:17.0.4.1_1-jdk-focal AS jre-build
 # for now we include the full module path to maintain compatibility
 # while still saving space (approx 200mb from the full distribution)
 RUN jlink \
-  --add-modules ALL-MODULE-PATH \
-  --strip-java-debug-attributes \
-  --no-man-pages \
-  --no-header-files \
-  --compress=2 \
-  --output /javaruntime
+         --add-modules ALL-MODULE-PATH \
+         --strip-java-debug-attributes \
+         --no-man-pages \
+         --no-header-files \
+         --compress=2 \
+         --output /javaruntime
 
 FROM debian:bullseye-20221024 AS build
 
@@ -57,9 +57,9 @@ ARG AGENT_WORKDIR=/home/${user}/agent
 
 RUN apt-get update \
   && apt-get -y install \
-  git-lfs \
-  curl \
-  fontconfig \
+    git-lfs \
+    curl \
+    fontconfig \
   && curl --create-dirs -fsSLo /usr/share/jenkins/agent.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${VERSION}/remoting-${VERSION}.jar \
   && chmod 755 /usr/share/jenkins \
   && chmod 644 /usr/share/jenkins/agent.jar \

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -41,7 +41,6 @@ RUN jlink \
          --compress=2 \
          --output /javaruntime
 
-
 FROM debian:bullseye-20221024 as build
 
 ARG VERSION=3071.v7e9b_0dc08466

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -34,12 +34,12 @@ FROM eclipse-temurin:17.0.4.1_1-jdk-focal AS jre-build
 # for now we include the full module path to maintain compatibility
 # while still saving space (approx 200mb from the full distribution)
 RUN jlink \
-         --add-modules ALL-MODULE-PATH \
-         --strip-java-debug-attributes \
-         --no-man-pages \
-         --no-header-files \
-         --compress=2 \
-         --output /javaruntime
+  --add-modules ALL-MODULE-PATH \
+  --strip-java-debug-attributes \
+  --no-man-pages \
+  --no-header-files \
+  --compress=2 \
+  --output /javaruntime
 
 FROM debian:bullseye-20221024 AS build
 
@@ -57,9 +57,9 @@ ARG AGENT_WORKDIR=/home/${user}/agent
 
 RUN apt-get update \
   && apt-get -y install \
-    git-lfs \
-    curl \
-    fontconfig \
+  git-lfs \
+  curl \
+  fontconfig \
   && curl --create-dirs -fsSLo /usr/share/jenkins/agent.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${VERSION}/remoting-${VERSION}.jar \
   && chmod 755 /usr/share/jenkins \
   && chmod 644 /usr/share/jenkins/agent.jar \
@@ -80,17 +80,17 @@ VOLUME /home/${user}/.jenkins
 VOLUME ${AGENT_WORKDIR}
 WORKDIR /home/${user}
 
+LABEL \
+  org.opencontainers.image.vendor="Jenkins project" \
+  org.opencontainers.image.title="Official Jenkins Agent Base Docker image" \
+  org.opencontainers.image.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" \
+  org.opencontainers.image.version="${VERSION}" \
+  org.opencontainers.image.url="https://www.jenkins.io/" \
+  org.opencontainers.image.source="https://github.com/jenkinsci/docker-agent" \
+  org.opencontainers.image.licenses="MIT"
+
 FROM build AS default
 COPY --from=jre-build /javaruntime $JAVA_HOME
 
 FROM build AS arm32
 COPY --from=arm32-jre-build /javaruntime $JAVA_HOME
-
-LABEL \
-    org.opencontainers.image.vendor="Jenkins project" \
-    org.opencontainers.image.title="Official Jenkins Agent Base Docker image" \
-    org.opencontainers.image.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" \
-    org.opencontainers.image.version="${VERSION}" \
-    org.opencontainers.image.url="https://www.jenkins.io/" \
-    org.opencontainers.image.source="https://github.com/jenkinsci/docker-agent" \
-    org.opencontainers.image.licenses="MIT"

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -22,11 +22,11 @@
 
 # To avoid "jmods: Value too large for defined data type" error,
 # Check docker bake target debian_jdk17_arm32
-# Cannot get jlink to properly work, in the v7 environment.  The jmods error
-# above gets generated each time.  Better to have a larger image than 
-# incorrect java binaries in 32bit arm.
-FROM  eclipse-temurin:17.0.4.1_1-jdk-focal AS arm32-jre-build
-RUN cp -r /opt/java/openjdk  /javaruntime
+# Cannot get jlink to properly work, in the v7 environment.
+# The jmods error above gets generated each time.
+# Better to have a larger image than incorrect java binaries in 32bit arm.
+FROM eclipse-temurin:17.0.4.1_1-jdk-focal AS arm32-jre-build
+RUN cp -r /opt/java/openjdk /javaruntime
 
 FROM eclipse-temurin:17.0.4.1_1-jdk-focal AS jre-build
 

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -80,7 +80,7 @@ VOLUME /home/${user}/.jenkins
 VOLUME ${AGENT_WORKDIR}
 WORKDIR /home/${user}
 
-FROM build as default
+FROM build AS default
 COPY --from=jre-build /javaruntime $JAVA_HOME
 
 FROM build as arm32

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -83,7 +83,7 @@ WORKDIR /home/${user}
 FROM build AS default
 COPY --from=jre-build /javaruntime $JAVA_HOME
 
-FROM build as arm32
+FROM build AS arm32
 COPY --from=arm32-jre-build /javaruntime $JAVA_HOME
 
 LABEL \

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -80,13 +80,11 @@ VOLUME /home/${user}/.jenkins
 VOLUME ${AGENT_WORKDIR}
 WORKDIR /home/${user}
 
-
 FROM build as default
 COPY --from=jre-build /javaruntime $JAVA_HOME
 
 FROM build as arm32
 COPY --from=arm32-jre-build /javaruntime $JAVA_HOME
-
 
 LABEL \
     org.opencontainers.image.vendor="Jenkins project" \

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -41,7 +41,7 @@ RUN jlink \
          --compress=2 \
          --output /javaruntime
 
-FROM debian:bullseye-20221024 as build
+FROM debian:bullseye-20221024 AS build
 
 ARG VERSION=3071.v7e9b_0dc08466
 ARG user=jenkins

--- a/17/bullseye/Dockerfile
+++ b/17/bullseye/Dockerfile
@@ -21,7 +21,14 @@
 #  THE SOFTWARE.
 
 # To avoid "jmods: Value too large for defined data type" error,
-FROM --platform=$BUILDPLATFORM eclipse-temurin:17.0.4.1_1-jdk-focal AS jre-build
+# Check docker bake target debian_jdk17_arm32
+# Cannot get jlink to properly work, in the v7 environment.  The jmods error
+# above gets generated each time.  Better to have a larger image than 
+# incorrect java binaries in 32bit arm.
+FROM  eclipse-temurin:17.0.4.1_1-jdk-focal AS arm32-jre-build
+RUN cp -r /opt/java/openjdk  /javaruntime
+
+FROM eclipse-temurin:17.0.4.1_1-jdk-focal AS jre-build
 
 # Generate smaller java runtime without unneeded files
 # for now we include the full module path to maintain compatibility
@@ -34,7 +41,8 @@ RUN jlink \
          --compress=2 \
          --output /javaruntime
 
-FROM debian:bullseye-20221024
+
+FROM debian:bullseye-20221024 as build
 
 ARG VERSION=3071.v7e9b_0dc08466
 ARG user=jenkins
@@ -64,7 +72,6 @@ ENV LANG C.UTF-8
 
 ENV JAVA_HOME=/opt/java/openjdk
 ENV PATH "${JAVA_HOME}/bin:${PATH}"
-COPY --from=jre-build /javaruntime $JAVA_HOME
 
 USER ${user}
 ENV AGENT_WORKDIR=${AGENT_WORKDIR}
@@ -73,6 +80,14 @@ RUN mkdir /home/${user}/.jenkins && mkdir -p ${AGENT_WORKDIR}
 VOLUME /home/${user}/.jenkins
 VOLUME ${AGENT_WORKDIR}
 WORKDIR /home/${user}
+
+
+FROM build as default
+COPY --from=jre-build /javaruntime $JAVA_HOME
+
+FROM build as arm32
+COPY --from=arm32-jre-build /javaruntime $JAVA_HOME
+
 
 LABEL \
     org.opencontainers.image.vendor="Jenkins project" \

--- a/docker-bake.hcl
+++ b/docker-bake.hcl
@@ -140,8 +140,6 @@ target "debian_jdk11" {
   platforms = ["linux/amd64", "linux/arm64", "linux/arm/v7", "linux/s390x"]
 }
 
-
-
 target "debian_jdk17" {
   dockerfile = "17/bullseye/Dockerfile"
   context = "."

--- a/docker-bake.hcl
+++ b/docker-bake.hcl
@@ -145,8 +145,9 @@ target "debian_jdk17" {
   context = "."
   args = {
     VERSION = REMOTING_VERSION,
-    target = "default"
+    
   }
+  target = "default"
   tags = [
     equal(ON_TAG, "true") ? "${REGISTRY}/${JENKINS_REPO}:${REMOTING_VERSION}-${BUILD_NUMBER}-jdk17": "",
     "${REGISTRY}/${JENKINS_REPO}:bullseye-jdk17",
@@ -162,8 +163,9 @@ target "debian_jdk17_arm32" {
   context = "."
   args = {
     VERSION = REMOTING_VERSION,
-    target = "arm32"
+    
   }
+  target = "arm32"
   tags = [
     equal(ON_TAG, "true") ? "${REGISTRY}/${JENKINS_REPO}:${REMOTING_VERSION}-${BUILD_NUMBER}-jdk17": "",
     "${REGISTRY}/${JENKINS_REPO}:bullseye-jdk17",

--- a/docker-bake.hcl
+++ b/docker-bake.hcl
@@ -5,6 +5,7 @@ group "linux" {
     "archlinux_jdk11",
     "debian_jdk11",
     "debian_jdk17",
+    "debian_jdk17_arm32"
   ]
 }
 
@@ -18,7 +19,7 @@ group "linux-arm64" {
 group "linux-arm32" {
   targets = [
     "debian_jdk11",
-    "debian_jdk17",
+    "debian_jdk17_arm32"
   ]
 }
 
@@ -139,11 +140,14 @@ target "debian_jdk11" {
   platforms = ["linux/amd64", "linux/arm64", "linux/arm/v7", "linux/s390x"]
 }
 
+
+
 target "debian_jdk17" {
   dockerfile = "17/bullseye/Dockerfile"
   context = "."
   args = {
-    VERSION = REMOTING_VERSION
+    VERSION = REMOTING_VERSION,
+    target = "default"
   }
   tags = [
     equal(ON_TAG, "true") ? "${REGISTRY}/${JENKINS_REPO}:${REMOTING_VERSION}-${BUILD_NUMBER}-jdk17": "",
@@ -152,5 +156,22 @@ target "debian_jdk17" {
     "${REGISTRY}/${JENKINS_REPO}:latest-bullseye-jdk17",
     "${REGISTRY}/${JENKINS_REPO}:latest-jdk17",
   ]
-  platforms = ["linux/amd64", "linux/arm64", "linux/arm/v7"]
+  platforms = ["linux/amd64", "linux/arm64"]
+}
+
+target "debian_jdk17_arm32" {
+  dockerfile = "17/bullseye/Dockerfile"
+  context = "."
+  args = {
+    VERSION = REMOTING_VERSION,
+    target = "arm32"
+  }
+  tags = [
+    equal(ON_TAG, "true") ? "${REGISTRY}/${JENKINS_REPO}:${REMOTING_VERSION}-${BUILD_NUMBER}-jdk17": "",
+    "${REGISTRY}/${JENKINS_REPO}:bullseye-jdk17",
+    "${REGISTRY}/${JENKINS_REPO}:jdk17",
+    "${REGISTRY}/${JENKINS_REPO}:latest-bullseye-jdk17",
+    "${REGISTRY}/${JENKINS_REPO}:latest-jdk17",
+  ]
+  platforms = ["linux/arm/v7"]
 }

